runners/flink/1.9/src/main/java/org/apache/beam/runners/flink/translation/types/EncodedValueSerializer.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink.translation.types;
+
+import java.io.IOException;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+import org.apache.flink.api.common.typeutils.TypeSerializerSchemaCompatibility;
+import org.apache.flink.api.common.typeutils.TypeSerializerSnapshot;
+import org.apache.flink.api.common.typeutils.base.TypeSerializerSingleton;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+
+/** {@link TypeSerializer} for values that were encoded using a {@link Coder}. */
+public final class EncodedValueSerializer extends TypeSerializerSingleton<byte[]> {
```

Comment:
Duplicate code, this class already exists twice. Please explain/document.

Reply:
This is due to the current structure of the flink runner. I can not simply reuse the code from the previous runner versions, because if I need to make a change to at least one of the files, it doesn't compile due to class duplicates.

Reply:
If these are the same files as for the 1.8, we can simply use the 1.8 version by adjusting the source path in `build.gradle`:

```gradle
  main_source_dirs = ["$basePath/src/main/java", "$basePath/1.8/src/main/java"]
```